src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5989,6 +5989,12 @@ def __call__(
             input.validate(params, tuple(["input"]))
         # Run the step
         result = self._handler(step_local_data.initialized_object, params)
+        if not isinstance(result, tuple):
+            raise BadArgumentException(
+                "The step returned type {};".format(type(result))
+                + "  expected a tuple with two values: output ID string"
+                  " and a step-specific value."
+            )
```

Comment:
Having the type info is a huge improvement.  Having the value would be a nice one-plus, unless you are concerned that it would somehow make the message harder to find and read.

In Dustin's case, it would have been illuminating to know up-front that the reason that the step failed was related to the failure of the `pcp3csv` tool.